### PR TITLE
feat(parsing): auto-create tailoring on RHEL minor version upgrade

### PR DESCRIPTION
**Pull request by RoamingNoMaD**
https://redhat.atlassian.net/browse/RHINENG-28323

The new Tailoring will use the original profile's default value definitions. It's not the best solution, but figuring out differences between override sets of profile versions is out of the scope of this task.

- Automatically create and reuse tailorings for supported RHEL minor versions.
- Apply version-specific profiles and value overrides.
- Audit-log new tailoring creation.
- Enforce unique tailorings per policy and OS minor version.
- Add tests for creation, reuse, logging, unsupported versions, and duplicate cleanup.

### CODE REVIEW

File: app/services/concerns/xccdf/tailorings.rb [hunk 1]
```diff
@@ -26,5 +21,22 @@ def tailored_profile
 
       @tailored_profile ||= tailoring.profile
     end
+
+    private
+
+    def find_or_create_tailoring
+      os_minor = @system.os_minor_version.to_i
+
+      ::Tailoring.find_or_create_by!(policy: @policy, os_minor_version: os_minor) do |tailoring|
```
[the remark this answers is not in the document]
Comment (RoamingNoMaD) [reply app/services/concerns/xccdf/tailorings.rb]: Valid, I'll need to clean the db before this can be merged.

**Comment by RoamingNoMaD** [comment]
Some cleanup needs to happen in stage/prod, but otherwise this is good to go.

The only failing test is due to the difference in how we now consume Kafka messages, unrelated to this PR.
(Basically batch consuming waits a bit (100 ms) to fill up a batch, then consumes it. If the tests don't wait that long or if there are other messages getting processed from the batch first, the test fails. We'll need to change the expectations to count with this.)

**Comment by romanblanco** [comment]
Ref. for future investigation: [RHINENG-28323](https://redhat.atlassian.net/browse/RHINENG-28323)

[RHINENG-28323]: https://redhat.atlassian.net/browse/RHINENG-28323?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Review by romanblanco — APPROVED** [review]
LGTM 👍
